packages/wdio-xvfb/src/index.ts

```diff
@@ -0,0 +1,197 @@
+import { promisify } from 'node:util'
+import { exec } from 'node:child_process'
+import os from 'node:os'
+import logger from '@wdio/logger'
+
+export interface XvfbOptions {
+    /**
+     * Force Xvfb to run even on non-Linux systems (for testing)
+     */
+    force?: boolean;
+}
+
+const execAsync = promisify(exec)
+
+export class XvfbManager {
+    private force: boolean
+    private log: ReturnType<typeof logger>
+
+    constructor(options: XvfbOptions = {}) {
+        this.force = options.force ?? false
+        this.log = logger('@wdio/xvfb')
+    }
+
+    /**
+     * Check if Xvfb should run on this system
+     */
+    shouldRun(): boolean {
+        if (this.force) {
+            return true
+        }
+
+        // Only run on Linux systems
+        if (os.platform() !== 'linux') {
+            return false
+        }
+
+        // Check if we're in a headless environment (no DISPLAY set or in CI)
+        const hasDisplay = process.env.DISPLAY
+        const isCI = !!(
+            process.env.CI ||
+            process.env.GITHUB_ACTIONS ||
+            process.env.JENKINS_URL ||
+            process.env.TRAVIS ||
+            process.env.CIRCLECI ||
+            process.env.GITLAB_CI ||
+            process.env.BUILDKITE ||
+            process.env.APPVEYOR
+        )
```

Comment:
Maybe we should just use https://www.npmjs.com/package/is-ci or similar?

Reply:
Hmmm thought I already had but I think it got lost in an update.